bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -363,7 +365,21 @@ private static void startHttpClient(HttpClient client) {
      *         not an image
      */
     public static RawType downloadImage(String url) {
-        return downloadImage(url, true, -1);
+        return downloadImage(url, DEFAULT_TIMEOUT);
+    }
+
+    /**
+     * Download the image data from an URL.
+     *
+     * If content type is not found in the headers, the data is scanned to determine the content type.
+     *
+     * @param url the URL of the image to be downloaded
+     * @param timeout the socket timeout to wait for data
+     * @return a RawType object containing the image, null if the content type could not be found or the content type is
+     *         not an image
+     */
+    public static RawType downloadImage(String url, long timeout) {
```

Comment:
It might make the API a little bit confusing if the second parameter is sometimes a boolean `scanTypeInContent ` and sometime the timeout.

Reply:
What do you propose considering that I don't want to break the current API ? To add only the new method with four arguments, the last one being the timeout ?

Reply:
this one is not breaking but might make it less easy to use, but I still intend to work on https://github.com/eclipse/smarthome/issues/4257 so maybe only introduce it in indeed in the 4 parameter version. It is up to you and the maintainers here :-)